dotnet/src/AutoGen.SemanticKernel/SemanticKernelAgent.cs

```diff
@@ -70,9 +72,7 @@ public async Task<IMessage> GenerateReplyAsync(IEnumerable<IMessage> messages, G
         CancellationToken cancellationToken = default)
     {
         var chatHistory = BuildChatHistory(messages);
-        var option = BuildOption(options);
-        var chatService = _kernel.GetRequiredService<IChatCompletionService>();
-        var response = chatService.GetStreamingChatMessageContentsAsync(chatHistory, option, _kernel, cancellationToken);
+        var response = _chatCompletionAgent.InvokeAsync(chatHistory, cancellationToken);
```

Comment:
From the implementation of [`_chatCompletionAgent.InvokeAsync`](https://github.com/microsoft/semantic-kernel/blob/092122390d5e7f6641517c175d26936ff45b7747/dotnet/src/Agents/Core/ChatCompletionAgent.cs#L25) it seems that `ChatCompletionAgent` doesn't use streaming-version api. This makes me wonder if the `ChatCompletionAgent` is the correct mapping to `SemanticKernelAgent`.

Reply:
I took your advice from our other conversation thread and implemented an `SemanticKernelChatCompletionAgent : IAgent` , a non streaming variant. 

I reverted the changes from `SemanticKernelAgent`,  added unit tests for `SemanticKernelChatCompletionAgent`. One of the unit tests: `SkChatCompletionAgentPluginTestAsync` required a middleware.

I introduced a `SkSequentialChatMessageContentConnector` which is a non streaming connector. The other `SemanticKernelChatMessageContentConnector` is a streaming version. I refactored the duplicate and moved it into the base class.

Let me know if anything is confusing. I'll update it, big thanks again for the advice !

Reply:
> I introduced a SkSequentialChatMessageContentConnector which is a non streaming connector. The other SemanticKernelChatMessageContentConnector is a streaming version. I refactored the duplicate and moved it into the base class.

I feel like it's unnecessary? Because `IStreamingMiddleware` is also an `IMiddleware`, so you can register `SemanticKernelChatMessageContentConnector` to the new `SemanticKernelChatCompletionAgent` as well

Reply:
Yes you are right, makes sense... I goofed and used `RegisterStreamingMiddleware` which caused me to create the refactor. Will revert.

Reply:
Resolving this, reverted and took your suggestion 

> You can register an IStreamingMiddleware to an IAgent because IStreamingMiddleware is also an IMiddleware